src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -45,15 +45,17 @@ public void addFolder(SpoonFolder o) {
 
 	@Override
 	public List<SpoonFile> getAllFiles() {
-		List<SpoonFile> result = new ArrayList<>();
+		Set<SpoonFile> result = new HashSet<>();
 
 		for (SpoonFile f : getFiles()) {
 			// we take care not to add a file that was already found in a folder
 			if (!result.contains(f)) {
 				result.add(f);
 			}
 		}
-		return result;
+
+		List<SpoonFile> resultAsList = new ArrayList<>(result);
+		return resultAsList;
```

Comment:
I recommend to avoid unnecessary local variable:

```
List<SpoonFile> resultAsList = new ArrayList<>(result);
return resultAsList;
```

Just return what is needed:
```
return new ArrayList<>(result);
```
It is shorter and easier to understand.